### PR TITLE
Local Site Permission Management

### PR DESCRIPTION
**Pull request by xuanrongzi**
Can add site to DB by password, and survey page read site and circle without Internet

### CODE REVIEW

File: www/AvailableSite.html [hunk 1]
```diff
@@ -32,30 +32,22 @@
...
 
-<div class="header navbar-default navbar-fixed-bottom">
-    <!--<div class="container-fluid" align="center" id="NewSurvey">-->
-    <div class="newsite" align = center>Add New Site</div>
-</div>
+
 
 <div class = "background">
+    <select id="site">
+			<option value ="default" selected disabled hidden>Please select a site</option>
```
Comment (NickWu007) [inline www/AvailableSite.html]: Please fix the indent here.

File: www/js/AvailableSite.js [hunk 2]
```diff
@@ -1,41 +1,179 @@
...
+    db.transaction(function(tx){
+        tx.executeSql('select distinct name, password, userId from USER', [], function(tx, rs){
+            if (rs.rows.length > 0) stored_user_info=rs.rows.item(0);
+        });
+        }, function(error){
+            alert("Transaction Error: "+error.message);
+        }, function() {
+            alert("successfully retrieved cached user info.");
+
+    });*/
```
Comment (NickWu007) [inline www/js/AvailableSite.js]: This section can be deleted.

File: www/js/AvailableSite.js [hunk 3]
```diff
@@ -1,41 +1,179 @@
...
+		data: JSON.stringify({
+			"action": "checkSitePassword",
+			"siteID": siteID,
+			"sitePasswordCheck": password
+		}),
+		success: function(passwordCheckResult){
+			//navigator.notification.alert(passwordCheckResult.validSitePassword);
+			//If site password is correct, submit survey
+			if(passwordCheckResult.validSitePassword == 1){
+				//navigator.notification.alert("Site password correct");
```
Comment (NickWu007) [inline www/js/AvailableSite.js:103]: use === to ensure the same type

File: www/js/AvailableSite.js [hunk 4]
```diff
@@ -1,41 +1,179 @@
...
+        }, function() {
+            alert("successfully Cleared all the information in Table SITE");
+            retriveSitePermission();
+
+    });
+    });
+});
+//CREATE TABLE IF NOT EXISTS SITE (siteId, siteName, userId, circle, state)
+function pushSitetoDB(siteID){
+        //get # of circle of specific site
```
Comment (NickWu007) [inline www/js/AvailableSite.js:131]: The name here is a bit misleading. It seems like you are trying to get the number of circles pushed in, so the name should actually be 'getCircleCount'

File: www/js/input-page.js [hunk 5]
```diff
@@ -90,6 +107,9 @@ function onDeviceReady(){
 			);
 		}
 	}, false);
+
```
Comment (NickWu007) [inline www/js/input-page.js]: unnecessary. Please delete.

File: www/js/input-page.js [hunk 6]
```diff
@@ -113,42 +133,39 @@ $( document ).ready(function() {
...
+                if(site_list.length>0){
+					var siteList = document.getElementById("site");
+                	for(var i=0; i<site_list.length; i++){
+                        var siteOption = document.createElement("option");
+						siteOption.text = site_list.item(i).siteName+"("+site_list.item(i).state+")";
+						siteOption.value = site_list.item(i).siteId;
+						siteList.add(siteOption);
+                	}
+					//alert("3");
+                }else{
```
Comment (NickWu007) [inline www/js/input-page.js:159]: Indentation here is really messed up. please fix

**Review by NickWu007 — CHANGES_REQUESTED** [review]

**Comment by xuanrongzi** [comment]
fixed

**Review by NickWu007 — APPROVED** [review]

**Pull request merged**